addons/website_event_exhibitor/views/event_templates_sponsor.xml

```diff
@@ -5,25 +5,17 @@
     <xpath expr="//div[@id='wrap']" position="inside">
         <div class="container mt32 mb16 d-none d-md-block d-print-none" t-if="event.sponsor_ids">
             <div t-attf-class="d-flex flex-wrap mb-5 #{'' if (len(event.sponsor_ids) > 10) else 'justify-content-md-center'}">
-                <t t-foreach="event.sponsor_ids" t-as="sponsor">
+                 <t t-foreach="event.sponsor_ids.sorted(lambda sponsor: not sponsor.website_published and sponsor.exhibitor_type != 'sponsor')" t-as="sponsor">
```

Comment:
Strange condition ... why filtering out published or "sponsor type" sponsors ?

Reply:
Yes right! The goal was to not display `Unpublished` ribbon to "sponsor type" sponsor because we do not display publish button for them in back-end, and they are assumed to be visible on website. The correct behavior would be, if someone changes the type of unpublished exhibitors to `sponsor`, we should publish them. The `onchange` is adapted accordingly.